staging/src/k8s.io/apiserver/pkg/server/lifecycle_signals.go

```diff
@@ -124,6 +124,11 @@ type lifecycleSignals struct {
 	// preshutdown hook(s) have finished running.
 	PreShutdownHooksStopped lifecycleSignal
 
+	// NotAcceptingNewRequest event is signaled when the server is no
+	// longer accpeting any new request, from this point on any new
+	// request will receive an error.
```

Comment:
```suggestion
	// longer accepting any new request, from this point on any new
	// request will receive an error.
```

we should add a test later asserting this behavior

Reply:
i think we have some assertion in the test now, can you review?

https://github.com/kubernetes/kubernetes/pull/110026/files#diff-bdc2ae2c7fb87165bf0c0984e74d807d88e24bd70c6f7071380981d672672329L469-L479

https://github.com/kubernetes/kubernetes/pull/110026/files#diff-bdc2ae2c7fb87165bf0c0984e74d807d88e24bd70c6f7071380981d672672329R273-R293